File: src/api/admin/microAppCategory.ts
import { post } from '@/utils/request'

// 获取分类列表
export function getList<T>(data: MicroAppCategory.GetListRequest) {
  return post<T>({
    url: '/admin/microAppCategory/getList',
    data,
  })
}

// 获取分类详情
export function getInfo<T>(data: MicroAppCategory.GetInfoRequest) {
  return post<T>({
    url: '/admin/microAppCategory/getInfo',
    data,
  })
}

// 创建分类
export function create<T>(data: MicroAppCategory.CreateRequest) {
  return post<T>({
    url: '/admin/microAppCategory/create',
    data,
  })
}

// 更新分类
export function update<T>(data: MicroAppCategory.UpdateRequest) {
  return post<T>({
    url: '/admin/microAppCategory/update',
    data,
  })
}

// 删除分类
export function deletes<T>(data: MicroAppCategory.DeletesRequest) {
  return post<T>({
    url: '/admin/microAppCategory/deletes',
    data,
  })
}

// 更新状态
export function updateStatus<T>(data: MicroAppCategory.UpdateStatusRequest) {
  return post<T>({
    url: '/admin/microAppCategory/updateStatus',
    data,
  })
}

// 获取启用的分类列表
export function getEnabledList<T>() {
  return post<T>({
    url: '/admin/microAppCategory/getEnabledList',
  })
}

File: service/router/admin/microAppCategory.go
package admin

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitMicroAppCategoryRouter(router *gin.RouterGroup) {
	categoryApi := api_v1.ApiGroupApp.ApiAdmin.MicroAppCategoryApi

	// 需要管理员权限的接口
	r := router.Group("", middleware.LoginInterceptor, middleware.AdminInterceptor)
	{
		r.POST("microAppCategory/getList", categoryApi.GetList)
		r.POST("microAppCategory/getInfo", categoryApi.GetInfo)
		r.POST("microAppCategory/create", categoryApi.Create)
		r.POST("microAppCategory/update", categoryApi.Update)
		r.POST("microAppCategory/deletes", categoryApi.Deletes)
		r.POST("microAppCategory/updateStatus", categoryApi.UpdateStatus)
	}

	// 公开接口（微应用详情页使用，无需登录）
	router.POST("microAppCategory/getEnabledList", categoryApi.GetEnabledList)
}
